docs/source/notes/distributed_training.rst

```diff
@@ -135,5 +136,160 @@ from the trainer.
+                Block(...) for _ in range(n_layers)
+            ]),
+            self.head = nn.Linear(...)
+
+        def forward(self, inputs):
+            ...
+
+        # FSDP Wrap Function
+        def fsdp_wrap_fn(self, module):
+            return isinstance(module, Block)
+
+        # Activation Checkpointing Function
+        def activation_checkpointing_fn(self, module):
+            return isinstance(module, Block)
+
+
+    class MyComposerModel(ComposerModel):
+
+        def __init__(self, n_layers):
+            super().__init__()
+            self.model = Model(n_layers)
+            ...
+
+        def forward(self, batch):
+            ...
+
+        def eval_forward(self, batch, outputs=None):
+            ...
+
+        def loss(self, outputs, batch):
+            ...
+
+        ...
+
+    composer_model = MyComposerModel(n_layers=3)
+
+    fsdp_config = {
+        'sharding_strategy': 'FULL_SHARD',
+        'min_params': 1e8,
+        'cpu_offload': False, # Not supported yet
+        'mixed_precision': 'DEFAULT',
+        'backward_prefetch': 'BACKWARD_POST',
+        'activation_checkpointing': False,
+        'activation_cpu_offload': False,
+        'verbose': True
+    }
+
+
+    trainer = Trainer(
+        model=composer_model,
+        fsdp_config=fsdp_config,
+        ...
+    )
+
+    trainer.fit()
+
+
+.. warning::
+    As of now now we don't support :code:`CPU Offloading` for FSDP.
+
+.. warning::
+    As of now, default parameters might not provide optimal convergence. Please proceed with caution.
+
+Composer's FSDP Auto Wrap Policy
+--------------------------------
+To make auto-wrapping easier on users, Composer uses a custom auto wrap policy that wraps modules according to the following rules:
+
+1) If any module is attributed with :code:`module._fsdp_wrap = True | False`, that choice will be respected.
+2) If the root module (e.g. `GPT`) defines a function :code:`def fsdp_wrap_fn(module: torch.nn.Module) -> bool`, then that function will be used to evaluate the root module's children.
+3) If any module has more parameters than :code:`fsdp_config['min_params']`, it will be wrapped.
+
+These rules are meant to make it easy for users to modify existing models for usage with FSDP. You can either add attributes to modules you want to wrap (#1), define a filter (#2), or make no changes at all and just use the size-based policy via :code:`fsdp_config['min_params'] = ...` (#3).
+
+In `gpt.py <https://github.com/mosaicml/benchmarks/blob/6972fe3000d5a5480d8757ff710965514155e8db/llm/llm/gpt.py>`__, you can see that `we used rule #2 <https://github.com/mosaicml/benchmarks/blob/6972fe3000d5a5480d8757ff710965514155e8db/llm/llm/gpt.py#L172>` to specify that all :code:`GPTBlock` modules within :code:`GPT` should be wrapped. Alternatively, we could have easily attributed each of the blocks with :code:`block._fsdp_wrap = True` and it would have accomplished the same thing. Whatever style you prefer, it's up to you!
```

Comment:
@bcui19 the link for "we used rule #2" is broken here, missing a "__" after the URL